code/sarix_model/sarix_model.py

```diff
@@ -37,580 +39,82 @@
+        pd.DataFrame(pred_qs[i, :, :]) \
+        .set_axis(df_hhs_last_obs['location'], axis='index') \
+        .set_axis(np.arange(1, max_horizon+1), axis='columns') \
+        .assign(output_type_id = q_label) \
+        for i, q_label in enumerate(q_labels)
+    ]) \
     .reset_index() \
     .melt(['location', 'output_type_id'], var_name='horizon') \
     .merge(df_hhs_last_obs, on='location', how='left')
-  
-  # build data frame with predictions on the original scale
-  preds_df['value'] = (preds_df['value'] + preds_df['inc_4rt_center_factor']) * preds_df['inc_4rt_scale_factor']
-  if transform == '4rt':
-    preds_df['value'] = np.maximum(preds_df['value'], 0.0) ** 4
-  else:
-    preds_df['value'] = np.maximum(preds_df['value'], 0.0) ** 2
-  
-  preds_df['value'] = (preds_df['value'] - 0.01 - 0.75**4) * preds_df['pop'] / 100000
-  preds_df['value'] = np.maximum(preds_df['value'], 0.0)
-
-  # keep just required columns and rename to match hub format
-  preds_df = preds_df[['location', 'wk_end_date', 'horizon', 'output_type_id', 'value']]
-
-  preds_df['target_end_date'] = preds_df['wk_end_date'] + pd.to_timedelta(7*preds_df['horizon'], unit='days')
-  preds_df['reference_date'] = ref_date
-  preds_df['horizon'] = preds_df['horizon'] - 2
-  preds_df['output_type'] = 'quantile'
-  preds_df['target'] = 'wk inc flu hosp'
-  preds_df.drop(columns='wk_end_date', inplace=True)
-  
-  if not Path(f'../../submissions-hub/model-output/UMass-sarix_{transform}').exists():
-    Path(f'../../submissions-hub/model-output/UMass-sarix_{transform}').mkdir(parents=True)
-
-  preds_df.to_csv(f'../../submissions-hub/model-output/UMass-sarix_{transform}/{str(ref_date)}-UMass-sarix_{transform}.csv', index=False)
-
+    
+    # build data frame with predictions on the original scale
+    preds_df['value'] = (preds_df['value'] + preds_df['inc_trans_center_factor']) * preds_df['inc_trans_scale_factor']
+    if power_transform == '4rt':
+        preds_df['value'] = np.maximum(preds_df['value'], 0.0) ** 4
+    else:
+        preds_df['value'] = np.maximum(preds_df['value'], 0.0) ** 2
+    
+    preds_df['value'] = (preds_df['value'] - 0.01 - 0.75**4) * preds_df['pop'] / 100000
+    preds_df['value'] = np.maximum(preds_df['value'], 0.0)
+    
+    # keep just required columns and rename to match hub format
+    preds_df = preds_df[['location', 'wk_end_date', 'horizon', 'output_type_id', 'value']]
+    
+    preds_df['target_end_date'] = preds_df['wk_end_date'] + pd.to_timedelta(7*preds_df['horizon'], unit='days')
+    preds_df['reference_date'] = ref_date
+    preds_df['horizon'] = preds_df['horizon'] - 2
+    preds_df['output_type'] = 'quantile'
+    preds_df['target'] = 'wk inc flu hosp'
+    preds_df.drop(columns='wk_end_date', inplace=True)
+    
+    if not Path(f'../../retrospective-hub/model-output/UMass-sarix_{power_transform}').exists():
+        Path(f'../../retrospective-hub/model-output/UMass-sarix_{power_transform}').mkdir(parents=True)
```

Comment:
you can pass `exist_ok=True` to `mkdir` if you want a one-liner way to create a directory that may or may not exist

Reply:
this actually got fixed up in the next commit, one of those recommendations for commit-by-commit reviews gone awry